crates/bindings-csharp/BSATN.Codegen/Type.cs

```diff
@@ -89,16 +100,16 @@ public BaseTypeDeclaration(GeneratorAttributeSyntaxContext context)
             Kind = TypeKind.Sum;
 
             if (
-                type.BaseType.TypeArguments[0]
+                type.BaseType.TypeArguments.FirstOrDefault()
```

Comment:
I wonder if there's a lint for this, not using `[0]`.

Reply:
Can't find anything at the moment.

Reply:
FYI `[0]` is not the same as `FirstOrDefault()`, and, unless you know optimization context of target platform, `[0]` can be considerably faster

Reply:
Yeah but this is not about speed, but about actually wanting the semantics of `FirstOrDefault()` where it returns nullable instead of throwing if there is no element. We could, of course, instead do manual length check + `[0]` access, but it shouldn't matter in practice, and this API is a bit cleaner.